### PR TITLE
Generate: remove flakyness in `test_generate_from_inputs_embeds_decoder_only`

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

`test_generate_from_inputs_embeds_decoder_only` had low probability of failure... but not zero (see #33593 and its CI runs)

Instead of assuming that calling `generate` with random `inputs_embeds` will result in a different text, let's check the scores instead. Even if the output text is the same (unlikely), the token scores are almost guaranteed to be different.

### CODE REVIEW

**Comment by gante** [comment]
hah, a PR to fight flakiness got stuck in another flaky test 🙃

File: tests/generation/test_utils.py [hunk 1]
```diff
@@ -1647,26 +1647,42 @@ def test_generate_from_inputs_embeds_decoder_only(self):
...
+                input_ids,
+                inputs_embeds=inputs_embeds,
+                max_new_tokens=5,
+                return_dict_in_generate=True,
+                output_scores=True,
+            )
+            self.assertListEqual(outputs_from_ids.sequences.tolist(), outputs_from_embeds.sequences.tolist())
 
-            # But if we pass different inputs_embeds, we should get different outputs
-            torch.manual_seed(0)
```
Comment (ydshieh) [inline tests/generation/test_utils.py:1659]: very nice to remove `torch.manual_seed`!

**Review by ydshieh — APPROVED** [review]
Great! I am up to merge as it is.

Another flaky test is worth another PR, and let's have fewer failures progressively.

But up to you :-)

**Comment by gante** [comment]
@ydshieh I don't have merge permissions, and I agree with you -- can you merge this PR for me? :)

**Pull request merged**